lib/Haskell/Law/Num/Nat.agda

```diff
@@ -0,0 +1,37 @@
+module Haskell.Law.Num.Nat where
+
+open import Haskell.Prim
+open import Haskell.Prim.Num
+
+open import Haskell.Law.Equality
+open import Haskell.Law.Num.Def
+
+-- Where to put this?
+-- Should we create some Nat.Laws / Nat.Properties / Law.Nat?
+addNat-suc : ∀ m n → m + suc n ≡ suc (m + n)
+addNat-suc zero    n = refl
+addNat-suc (suc m) n = cong suc (addNat-suc m n)
+
+module Termination where
+  Nat-+-idʳ : ∀ (x : Nat) → x + 0 ≡ x
+  Nat-+-idʳ zero    = refl
+  Nat-+-idʳ (suc x) = cong suc (Nat-+-idʳ x)
+
+open Termination
+
+instance
+  iLawfulNum : IsLawfulNum Nat
+  IsLawfulNum.+-assoc iLawfulNum = {!!}
+  IsLawfulNum.+-comm iLawfulNum zero y = sym (Nat-+-idʳ y)
```

Comment:
Style-wise, I would avoid inlining the proofs into the instance definition. We could instead give each property a top-level name, or we could define the whole record as a record type and put each of the fields as a definition in a `where` block.

Reply:
Sounds good to me. This will also resolve the termination problems I faced so far. :)